src/dom.ts

```diff
@@ -269,3 +269,67 @@ export function compareElementArea(a: HTMLElement, b: HTMLElement): number {
 
     return aArea - bArea
 }
+
+/** Computes a unique CSS selector for the current HTMLElement.
+ *  Currently buggy, returns a unique selector but doesn't always correspond
+ *  to the selector you should use for functions such as querySelectorAll().
+ */
+export function getSelector(elem: HTMLElement) {
+   if (elem === null || elem === undefined)
+      return undefined
+
+   if (elem == document.body || !elem.parentNode)
+      return "body"
+
+   if (elem.id)
+      return "#" + elem.id
+
+   let parent = elem.parentNode as HTMLElement
+   let n = [].indexOf.call(parent.childNodes, elem) + 1
+   return getSelector(parent) + " :nth-child(" + n + ")"
+}
+
+export const hintworthy_js_elems = {}
+
+/** Adds or removes an element from the hintworthy_js_elems array of the
+ *  current tab.
+ *  This function is made available to the page's. We should be very careful
+ *  about the input it accepts and what it does with it.
+ */
+export function registerEvListenerAction(action: string, ev: string): void {
+   if (!(this instanceof HTMLElement))
+      return
+
+   switch (ev) {
+      case "click":
+      case "hover":
+      case "mouseover":
```

Comment:
"hover" isn't a real event, is it?

Also check for "mousedown", "mouseup" as similar to click and "mouseenter", "mousemove", as similar to "mouseover" (hover).

*Maybe* check for "mouseout", "mouseleave" as similar to "mouseover".

Reply:
I can't find a list of possible events in the W3C's spec but I think you're right, hover doesn't exist. Including "mouseout", "mouseleave" and "mouseover" doesn't seem to break anything on the few websites I'm testing Tridactyl on so let's add them.